File: stacks/nodejs/nestjs/src/controllers/hash/hash.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { LoopStatus } from '../../models/loop-status';
import { HashService } from '../../services/hash/hash.service';
import { loopCall } from '../../utils';

@Controller()
export class HashController {
  constructor(private readonly hashService: HashService) {}

  @Get('/hash')
  async getStatus(
    @Query('delay') delayInput,
    @Query('percentage') percentageInput,
  ): Promise<LoopStatus> {
    return loopCall(delayInput, percentageInput, this.hashService.process);
  }
}

File: stacks/elixir/plug-cowboy/lib/plug_cowboy_app/controllers/hash_controller.ex
defmodule Controllers.HashController do
  import Services.Hash
  import Utils.DelayedTask
  alias Models.LoopStatus
  @moduledoc false

  def hash(percentage, delay) do
    times = start(&process/0, percentage, delay)
    LoopStatus.ok(times > 0, times)
  end

end
